### PR TITLE
LID: several random samples for long file

### PR DESCRIPTION
**Pull request by karpnv**
# What does this PR do ?

Use several random samples for long files

**Collection**: ASR

# Changelog 
Added parameters:
- segment_duration (float): random sample duration in seconds
- num_segments (int): number of segments of file to use for majority vote

# Usage

```python
lang_model = nemo_asr.models.EncDecSpeakerLabelModel.from_pretrained(model_name="langid_ambernet")
lang = lang_model.get_label(filename, segment_duration = 60*6, num_segments = 3)
```

**PR Type**:
- [V] New Feature

## Who can review?
@fayejf

### CODE REVIEW

File: nemo/collections/asr/models/label_models.py [hunk 1]
```diff
@@ -466,25 +468,74 @@ def infer_file(self, path2audio_file):
...
-            path2audio_file: path to audio wav file
+            path2audio_file (str): path to audio wav file
+            segment_duration (float): random sample duration in seconds
+            num_segments (int): number of segments of file to use for majority vote
 
         Returns:
             label: label corresponding to the trained model
         """
-        _, logits = self.infer_file(path2audio_file=path2audio_file)
+        audio, sr = librosa.load(path2audio_file, sr=None)
```
Comment (nithinraok) [inline nemo/collections/asr/models/label_models.py:511]: can you replace this with `sf.read` it is much faster

File: nemo/collections/asr/models/label_models.py [hunk 2]
```diff
@@ -466,25 +468,74 @@ def infer_file(self, path2audio_file):
...
+        return emb, logits
+
+    def get_label(self, path2audio_file: str, segment_duration: float = np.inf, num_segments: int = 1):
         """
         Returns label of path2audio_file from classes the model was trained on.
         Args:
-            path2audio_file: path to audio wav file
+            path2audio_file (str): path to audio wav file
+            segment_duration (float): random sample duration in seconds
+            num_segments (int): number of segments of file to use for majority vote
```
Comment (nithinraok) [inline nemo/collections/asr/models/label_models.py:506]: instead of num_segments, just do non-overlap segments from start to end based on 5 sec audio samples? Have you done ablation study on what is best?

Comment (karpnv) [reply nemo/collections/asr/models/label_models.py:506]: I personally didn't, but it was suggested by Fai. This is for very long audio (several hours). We take several segments and get result by majority vote

File: nemo/collections/asr/models/label_models.py (hunk 1 above)
Comment (karpnv) [reply nemo/collections/asr/models/label_models.py]: done

**Review by nithinraok — DISMISSED** [review]
LGTM, Please add a random seed for selection of random segments.

**Comment by karpnv** [comment]
added random_seed parameter

**Review by nithinraok — APPROVED** [review]
Thanks, LGTM

**Comment by nithinraok** [comment]
jenkins

**Review by nithinraok — APPROVED** [review]

**Comment by nithinraok** [comment]
jenkins

File: nemo/collections/asr/models/label_models.py (hunk 1 above)
Comment (liuspencersjtu) [reply nemo/collections/asr/models/label_models.py]: I know it might not have been originally designed to support reading mp3 or multi-channel (stereo) wav files, but it was able to do so in the past with librosa.load, however, it may result in errors after switching to sf.read. Should we consider adding support for more formats or stick to using librosa.load for consistency?
